bundles/org.openhab.binding.flume/src/main/java/org/openhab/binding/flume/internal/api/FlumeApi.java

```diff
@@ -0,0 +1,411 @@
+        verifyToken();
+
+        String url = APIURL_BASE + String.format(APIURL_GETCURRENTFLOWRATE, this.userId, deviceId);
+        Request request = httpClient.newRequest(url).method(HttpMethod.GET)
+                .header("Authorization", "Bearer " + accessToken).header("accept", "application/json");
+
+        logger.trace("GET: {}", url);
+        ContentResponse response = request.send();
+        JsonObject jsonResponse = validateResponse(response);
+
+        final FlumeApiCurrentFlowRate[] currentFlowRates = gson.fromJson(jsonResponse.get("data").getAsJsonArray(),
+                FlumeApiCurrentFlowRate[].class);
+
+        if (currentFlowRates == null) {
+            return null;
+        }
+        if (currentFlowRates.length < 1) {
+            return null;
+        }
+
+        return currentFlowRates[0];
+    }
+
+    public List<FlumeApiUsageAlert> fetchUsageAlerts(String deviceId, int limit)
+            throws FlumeApiException, IOException, InterruptedException, TimeoutException, ExecutionException {
+        verifyToken();
+
+        String url = APIURL_BASE
+                + String.format(APIURL_FETCHUSAGEALERTS, userId, deviceId, limit, "triggered_datetime", "DESC");
+        Request request = httpClient.newRequest(url).method(HttpMethod.GET)
+                .header("Authorization", "Bearer " + accessToken).header("accept", "application/json");
+
+        logger.trace("GET: {}", url);
+        ContentResponse response = request.send();
+        JsonObject jsonResponse = validateResponse(response);
+
+        final FlumeApiUsageAlert[] listUsageAlerts = gson.fromJson(jsonResponse.get("data").getAsJsonArray(),
+                FlumeApiUsageAlert[].class);
+
+        return Arrays.asList(listUsageAlerts);
+    }
+
+    public List<FlumeApiUsageAlert> fetchNotificatinos(String deviceId, int limit)
+            throws FlumeApiException, IOException, InterruptedException, TimeoutException, ExecutionException {
+        verifyToken();
+
+        String url = APIURL_BASE
+                + String.format(APIURL_FETCHNOTIFICATIONS, userId, deviceId, limit, "triggered_datetime", "DEC");
+        Request request = httpClient.newRequest(url).method(HttpMethod.GET)
+                .header("Authorization", "Bearer " + accessToken).header("accept", "application/json");
+
+        logger.trace("GET: {}", url);
+        ContentResponse response = request.send();
+        JsonObject jsonResponse = validateResponse(response);
+
+        final FlumeApiUsageAlert[] listUsageAlerts = gson.fromJson(jsonResponse.get("data").getAsJsonArray(),
+                FlumeApiUsageAlert[].class);
+
+        return Arrays.asList(listUsageAlerts);
+    }
+
+    private JsonObject validateResponse(ContentResponse response) throws FlumeApiException {
+        logger.trace("Response: {}", response.getContentAsString());
+
+        if (response.getStatus() == 429) {
+            throw new FlumeApiException("@text/api.rate-limit-exceeded", 429, false);
```

Comment:
Does the API add details to the response that explains when new requests are accepted again? If so it could be added to the message and or retry mechanism could be added.

Reply:
Unfortunately no.  However,  when the device goes offline due to a rate-limit, it continues to try to attempt at the next polling cycle and will go back online when it succeeds.